### PR TITLE
Refactor/authorization

### PR DESCRIPTION
**Pull request by macintoshpie**
#### What's this PR do?
Makes authorization more consistent

#### How should this be manually tested?

Data Quality Checks
DQ results are cached with a key that includes org ID now. Verify this still works by running DQ checks from inventory list page and the import file flow. Also test with a suborganization to make sure it works as well.

General functional testing
Poke around the entire site to test functionality is maintained. The largest concern is that a user that was previously authorized to perform an action no longer is. Note that you must test as a non-superuser for authorization to be properly tested

#### What are the relevant tickets?
#2379

### CODE REVIEW

File: seed/lib/superperms/orgs/decorators.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
     Organization,
     OrganizationUser
 )
+from seed.lib.superperms.orgs.permissions import get_org_id
```
Comment (macintoshpie) [inline seed/lib/superperms/orgs/decorators.py:20]: same method used by the OrgMixin, which is used for getting the org id inside of a view. This should be used to ensure each place is using the same strategy for getting the org id

File: seed/lib/superperms/orgs/permissions.py [hunk 2]
```diff
@@ -55,21 +55,43 @@ def get_org_or_id(dictlike):
 
 
 def get_org_id(request):
-    """Get org id from request"""
-    org_id = get_org_or_id(request.query_params)
-    if not org_id:
+    """Extract the ``organization_id`` regardless of HTTP method type."""
```
Comment (macintoshpie) [inline seed/lib/superperms/orgs/permissions.py:58]: updated method by merging with the _get_org_id func from decorators.py. Also tweaked the sequence of where we look for IDs.

File: seed/utils/api.py [hunk 3]
```diff
@@ -319,15 +319,16 @@ def get_organization(self, request, return_obj=False):
             if not org_id:
                 org = get_user_org(request.user)
                 org_id = int(getattr(org, 'pk'))
+            if not org:
```
Comment (macintoshpie) [inline seed/utils/api.py:322]: I can comment on this in person

File: seed/tests/test_labels.py [hunk 4]
```diff
@@ -106,10 +106,10 @@ def test_error_occurs_when_trying_to_apply_a_label_to_taxlotview_from_a_differen
         with transaction.atomic():
             with self.assertRaises(IntegrityError):
                 self.api_view.add_labels(
-                    self.api_view.models['taxlot'].objects.none(),
-                    'taxlot',
-                    [org_1_taxlot.id],
-                    [self.org_2_status_label.id]
+                    qs=self.api_view.models['taxlot'].objects.none(),
+                    inventory_type='taxlot',
+                    inventory_ids=[org_1_taxlotview.id],
```
Comment (macintoshpie) [inline seed/tests/test_labels.py:111]: The one important change is this line, we were passing canonical IDs when it needed to be view IDs.

Test was failing for me b/c I was using `--keepdb` flag when running test. Previously this passed b/c the IDs of views and canonical inventory were "synched". Same goes for the other label test I modified

File: seed/views/data_quality.py [hunk 5]
```diff
@@ -159,7 +159,7 @@ def csv(self, request, pk):
               required: true
               paramType: path
         """
-        data_quality_results = get_cache_raw(DataQualityCheck.cache_key(pk))
+        data_quality_results = get_cache_raw(DataQualityCheck.cache_key(pk, self.get_organization(request)))
```
Comment (macintoshpie) [inline seed/views/data_quality.py:162]: Added org id to dq cache key to protect against unauthorized access

File: seed/views/v3/label_inventories.py [hunk 6]
```diff
@@ -128,15 +137,23 @@ def label_factory(self, inventory_type, label_id, inventory_id):
...
-                self.label_factory(inventory_type, label_id, pk)
-                for label_id in add_label_ids for pk in inventory_ids
-                if pk not in exclude[label_id]
-            ]
+            new_inventory_labels = []
+            for label_id in add_label_ids:
+                for pk in inventory_ids:
+                    if pk not in exclude[label_id]:
+                        new_inventory_label = self.label_factory(inventory_type, label_id, pk)
+                        new_inventory_labels.append(new_inventory_label)
```
Comment (macintoshpie) [inline seed/views/v3/label_inventories.py:156]: There should be no functional change here. Just reorganized when I was debugging a test to help better understand what was happening

File: seed/views/v3/import_files.py [hunk 7]
```diff
@@ -157,15 +157,15 @@ def retrieve(self, request, pk=None):
         })
 
     @swagger_auto_schema_org_query_param
-    @has_perm_class('requires_viewer')
     @api_endpoint_class
     @ajax_request_class
+    @has_perm_class('requires_viewer')
```
Comment (macintoshpie) [inline seed/views/v3/import_files.py:162]: has_perm_class must come "after" the other decorators, ie it must be wrapped itself by the other decorators. This is necessary b/c the api_endpoint_class decorator needs to configure the request.user before we check authz

**Review by adrian-lara — COMMENTED** [review]
Code changes look good - a few very small tweaks. Wanted to leave drop the comments while I try some manual tests.

**Review by adrian-lara — APPROVED** [review]
Manual testing looks good to me! We should get this into development and dev1 sooner so that overall app functionality can be tested while other tickets are tested.

**Pull request merged**